src/d3a/d3a_core/export.py

```diff
@@ -169,48 +174,106 @@ def move_root_plot_folder(self):
             shutil.move(os.path.join(old_dir, si), self.plot_dir)
         shutil.rmtree(old_dir)
 
-    def _export_area_with_children(self, area: Area, directory: dir, is_first: bool = False):
+    def _export_spot_markets_stats(self, area: Area, directory: dir, is_first: bool) -> None:
```

Comment:
I played around a lot with this in order to reduce the code repetition. However, I would have to add if/elif/else blocks to the potential universal function that would make it really ugly. Also, for the balancing markets we currently do not export the bids.

Reply:
Given your conversation, I think that these could actually be classes rather than functions. There would be a common interface (no universal function) and no need to use `if` statements, since each class would have its own implementation. Maybe this could be a way to proceed if you think it's suitable and you want to spend some time on it.

Reply:
To be dealt with as part of refactoring story: https://gridsingularity.atlassian.net/browse/D3ASIM-3600